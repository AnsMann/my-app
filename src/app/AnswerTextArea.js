import React, { useState } from 'react'
import styled from 'styled-components'

const Answer = styled.textarea`
  border: 1px solid #007fbf;
  border-radius: 10px;
  font-family: inherit;
  font-size: 1rem;
  padding: 15px 20px 0;
  text-align: justify;
  width: 100%;
`

export function AnswerTextArea({
  name,
  placeholder = 'Type here',
  value,
  rows = '5',
}) {
  const [inputValue, setInputValue] = useState('')
  return (
    <Answer
      onChange={event => setInputValue(event.target.value)}
      rows={rows}
      placeholder={placeholder}
      name={name}
      value={value || inputValue}
    />
  )
}
