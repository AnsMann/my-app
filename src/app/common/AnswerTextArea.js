import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'

const Answer = styled.textarea`
  border: 1px solid #007fbf;
  border-radius: 10px;
  font-family: inherit;
  font-size: 1rem;
  padding: 15px 20px 0;
  text-align: justify;
  width: 100%;
`

export function AnswerTextArea({
  name,
  placeholder = 'Type here',
  defaultValue,
  rows = '5',
}) {
  return (
    <Answer
      rows={rows}
      placeholder={placeholder}
      name={name}
      defaultValue={defaultValue}
    />
  )
}

AnswerTextArea.propTypes = {
  rows: PropTypes.number,
  placeholder: PropTypes.string,
  name: PropTypes.string,
  defaultValue: PropTypes.string,
}
