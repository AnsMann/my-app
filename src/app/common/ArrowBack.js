import React from 'react'
import styled from 'styled-components'
import { library } from '@fortawesome/fontawesome-svg-core'
import { faLongArrowAltLeft } from '@fortawesome/free-solid-svg-icons'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import PropTypes from 'prop-types'

library.add(faLongArrowAltLeft)

const Back = styled.div`
  align-items: center;
  background-color: white;
  border: solid 2px #007fbf;
  border-radius: 20px;
  color: #007fbf;
  display: flex;
  font-size: 3.5rem;
  height: 60px;
  justify-content: center;
  width: 80px;
`

export function ArrowBack({ onBackClick, history }) {
  return (
    <Back onClick={() => onBackClick(history)}>
      <FontAwesomeIcon icon={faLongArrowAltLeft} />
    </Back>
  )
}

ArrowBack.propTypes = {
  onBackClick: PropTypes.func.isRequired,
  history: PropTypes.object.isRequired,
}
