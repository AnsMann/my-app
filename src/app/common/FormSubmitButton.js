import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'

const SaveButton = styled.button`
  background: #007fbf;
  border-radius: 10px;
  color: #ffffff;
  font-size: 1rem;
  height: 30px;
  margin: 5px 0;
  width: 100%;
`

export function FormSubmitButton({ title }) {
  return <SaveButton>{title}</SaveButton>
}

FormSubmitButton.propTypes = {
  title: PropTypes.string,
}
