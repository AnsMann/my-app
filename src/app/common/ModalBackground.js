import React from 'react'
import styled from 'styled-components'

const Background = styled.div`
  background: white;
  height: 100vh;
  opacity: 0.5;
  position: absolute;
  top: 0px;
  left: 0px;
  width: 100vw;
  z-index: 100;
`

export function ModalBackground() {
  return <Background />
}
