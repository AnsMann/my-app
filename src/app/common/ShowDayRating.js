import React from 'react'
import PropTypes from 'prop-types'

export function ShowDayRating({ entryRating }) {
  return entryRating ? (
    <h3>
      This day was: <span>{evaluateRating(entryRating)}</span>
    </h3>
  ) : (
    <h3>No rating of this day</h3>
  )
}

function evaluateRating(rating) {
  const ratingMap = {
    1: 'ðŸ˜”',
    2: 'ðŸ˜¶',
    3: 'ðŸ˜ƒ',
  }
  return ratingMap[rating]
}

ShowDayRating.propTypes = {
  entryRating: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
}
