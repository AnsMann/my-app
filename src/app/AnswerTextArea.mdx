---
name: Textarea
---

import { Playground, Props } from 'docz'
import { AnswerTextArea } from './AnswerTextArea.js'

# Textarea

<Props of={AnswerTextArea} />

**This textarea is used to create or edit diary entries**

<Playground>
  <AnswerTextArea
    name={'name'}
    defaultValue={'defaultvalue instead of placeholder'}
  />
</Playground>
