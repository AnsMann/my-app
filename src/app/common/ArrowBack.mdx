---
name: ArrowBack Button
menu: Buttons
---

import { Playground, Props } from 'docz'
import { ArrowBack } from './ArrowBack.js'

# Back Button

<Props of={ArrowBack} />

**Button that is shown to return to previous page.**

<Playground>
  <ArrowBack
    onBackClick={() => window.history.back()}
    history={window.history}
  />
</Playground>
