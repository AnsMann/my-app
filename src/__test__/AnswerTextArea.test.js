import React from 'react'
import renderer from 'react-test-renderer'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { AnswerTextArea } from '../app/common/AnswerTextArea'

Enzyme.configure({ adapter: new Adapter() })

describe('AnswerTextArea for form with default values for rows and placeholder', () => {
  it('renders an element with required props', () => {
    const component = renderer.create(<AnswerTextArea name={'test title'} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders an element with props sizes', () => {
    const component = renderer.create(
      <AnswerTextArea rows={3} name={'test title'} />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders an element with props placeholder', () => {
    const component = renderer.create(
      <AnswerTextArea placeholder={'placeholder'} name={'test title'} />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders an element with props defaultvalue', () => {
    const component = renderer.create(
      <AnswerTextArea defaultValue={'default value'} name={'test title'} />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders an element with all props', () => {
    const component = renderer.create(
      <AnswerTextArea
        rows={4}
        placeholder={'placeholder'}
        name={'name'}
        defaultValue={'default value'}
      />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
