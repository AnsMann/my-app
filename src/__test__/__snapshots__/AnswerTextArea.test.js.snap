// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnswerTextArea for form with default values for rows and placeholder renders an element 1`] = `
<textarea
  className="AnswerTextArea__Answer-sc-1ca9gm9-0 ekQCHP"
  name="test title"
  placeholder="Type here"
  rows="5"
/>
`;

exports[`AnswerTextArea for form with default values for rows and placeholder renders an element with all props 1`] = `
<textarea
  className="AnswerTextArea__Answer-sc-1ca9gm9-0 ekQCHP"
  defaultValue="default value"
  name="name"
  placeholder="placeholder"
  rows={4}
/>
`;

exports[`AnswerTextArea for form with default values for rows and placeholder renders an element with props defaultvalue 1`] = `
<textarea
  className="AnswerTextArea__Answer-sc-1ca9gm9-0 ekQCHP"
  defaultValue="default value"
  name="test title"
  placeholder="Type here"
  rows="5"
/>
`;

exports[`AnswerTextArea for form with default values for rows and placeholder renders an element with props name 1`] = `
<textarea
  className="AnswerTextArea__Answer-sc-1ca9gm9-0 ekQCHP"
  name="name"
  placeholder="Type here"
  rows="5"
/>
`;

exports[`AnswerTextArea for form with default values for rows and placeholder renders an element with props placeholder 1`] = `
<textarea
  className="AnswerTextArea__Answer-sc-1ca9gm9-0 ekQCHP"
  name="test title"
  placeholder="placeholder"
  rows="5"
/>
`;

exports[`AnswerTextArea for form with default values for rows and placeholder renders an element with props sizes 1`] = `
<textarea
  className="AnswerTextArea__Answer-sc-1ca9gm9-0 ekQCHP"
  name="test title"
  placeholder="Type here"
  rows={3}
/>
`;
